### PR TITLE
refactor(Pill): deprecate Pill and use element from Palette

### PR DESCRIPTION
**Pull request by araujobarret**
### Description

This PR deprecates Pill implementation in Eigen and updates in 3 screens with the new `Pill` from Palette.

VIsual Changes:
- Saved Alert screen has no difference
- Create Alert screen has a difference, the border is lighter according to the DS when the element is not `disabled`
- Search Pills screen has a difference, the border is lighter according to the DS

| Before | After    |
| -------| -------|
|<video src="https://github.com/artsy/eigen/assets/15792853/6f7590fc-00d3-41f6-bf58-ce13de0b88d7"  />|              <video src="https://github.com/artsy/eigen/assets/15792853/28687fc6-d276-42c5-843e-b53ad876654a" />|
|  |  |
|<video src="https://github.com/artsy/eigen/assets/15792853/f8e12632-aa1f-422c-a4d8-c31af900a3b3" /> | <video src="https://github.com/artsy/eigen/assets/15792853/b2954c85-ea9a-4caf-bba1-a5c50e0ac982"  />|

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- refactor: update Pill in the screens: Saved Alert, Create Alert and Search - araujobarret

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bcadf9f2223df4cdfad6bb4c16086c9daf7fd980
</p>

**Review by gkartalis — APPROVED** [review]
Looks amazing! Thanks for pushing this to the finish line!!! 

Lets wait for Onyx people to review as well before merging!

File: src/app/Components/Pill/Pill.tsx [hunk 1]
```diff
@@ -70,6 +70,10 @@ const getSize = (size: PillSize): { height: number; paddingRight: number; paddin
   }
 }
 
+/**
+ * @deprecated
+ * Use `Image` from palette instead.
```
Comment (nickskalkin) [inline src/app/Components/Pill/Pill.tsx:75]: > Use `Image` from palette instead.

Should it be `Use "Pill" from palette instead`?

**Review by nickskalkin — APPROVED** [review]
Looks good to me, only left a tiny comment about deprecation message above

File: src/app/Components/Pill/Pill.tsx (hunk 1 above)
Comment (araujobarret) [reply src/app/Components/Pill/Pill.tsx:75]: ops, wrong copy/pasta 🍝 , good catch!

**Review by nickskalkin — APPROVED** [review]

**Review by anandaroop — APPROVED** [review]

**Pull request merged**